### PR TITLE
Fix main thread violation at plugin startup (#57) + compiler warnings

### PR DESCRIPTION
**Pull request by ettore**
- Fixes a run-time main thread violation when starting up the plugin, per discussion in issue #57 
- Also fixes a few compile-time warnings (Xcode 14.3.1)

### CODE REVIEW

File: ios/Plugin/NewRelicCapacitorPluginPlugin.swift [hunk 1]
```diff
@@ -156,16 +156,18 @@ public class NewRelicCapacitorPluginPlugin: CAPPlugin {
...
+        DispatchQueue.main.async {
+            if collectorAddress == nil && crashCollectorAddress == nil {
+                NewRelic.start(withApplicationToken: appKey)
+            } else {
+                NewRelic.start(withApplicationToken: appKey,
+                               andCollectorAddress: collectorAddress ?? "mobile-collector.newrelic.com",
+                               andCrashCollectorAddress: crashCollectorAddress ?? "mobile-crash.newrelic.com")
+            }
+
+            call.resolve()
```
Comment (ettore) [inline ios/Plugin/NewRelicCapacitorPluginPlugin.swift:169]: I moved the `resolve` call inside the main thread in order to know that we have really completed the whole job of starting up New Relic before telling Capacitor that we are done. This seems safer to me.

On the Capacitor side there doesn't seem to be any requirements on when to call `resolve` in their [official docs](https://capacitorjs.com/docs/plugins/ios#returning-data-back).  In particular they do not specify if `resolve` must be synchronous with the calling context.  NR initialization seems to work fine with `resolve` being inside the main thread closure or outside of it, so if there's a preference please let me know the reason and I can move it.

Comment (ndesai-newrelic) [reply ios/Plugin/NewRelicCapacitorPluginPlugin.swift:169]: If we are returning data from the main thread, we must call resolve from the main thread. However, you may call resolve from the outside as well without any issues.

**Review by ndesai-newrelic — APPROVED** [review]
thank you for your fix.

**Comment by ndesai-newrelic** [comment]
@ettore we will release this next week with other changes we made on native agents.

**Comment by ettore** [comment]
sounds great! Thank you for the quick merge.

**Pull request merged**